Rules/CommonScripts/Holiday/Christmas/Christmas.as

```diff
@@ -101,15 +102,33 @@ void onTick(CRules@ this)
 
 			bool is_spawned = false;
 
-			if (trees_blue.length > 0)
-			{
-				spawnPresent(trees_blue[XORRandom(trees_blue.length)].getPosition(), 0);
-				is_spawned = true;
-			}
-			if (trees_red.length > 0)
+			for (uint i = 0; i < gifts_per_hoho; i++)
 			{
-				spawnPresent(trees_red[XORRandom(trees_red.length)].getPosition(), 1);
-				is_spawned = true;
+				if (trees_blue.length > 0)
+				{
+					int random = XORRandom(trees_blue.length);
+					spawnPresent(trees_blue[random].getPosition(), XORRandom(8));
+					trees_blue.removeAt(random);
+					is_spawned = true;
```

Comment:
you're setting `is_spawned` to true 4 times in the for loop, and it will be triggered twice guaranteed, a bit much?

Reply:
follow-up: `is_spawned` will always be true due to parachute changes, so just remove the variable

Reply:
resolved